catalog-rest-service/src/main/resources/json/schema/type/storage.json

```diff
@@ -0,0 +1,99 @@
+{
+  "$id": "https://open-metadata.org/schema/entity/type/storage.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Table",
+  "description": "Defines different tier of storage classes offered by a Storage Service",
+  "type": "object",
+  "definitions": {
+    "storageServiceType": {
+      "description": "Type of storage service such as S3, GCS, HDFS...",
+      "type": "string",
+      "javaType": "org.openmetadata.catalog.type.StorageServiceType",
+      "enum": [
+        "S3",
+        "GCS",
+        "HDFS",
+        "ABFS"
+      ],
+      "javaEnums": [
+        {
+          "name": "S3"
+        },
+        {
+          "name": "GCS"
+        },
+        {
+          "name": "HDFS"
+        },
+        {
+          "name": "ABFS"
+        }
+      ]
+    },
+    "storageClassType": {
+      "description": "Type of storage class offered by the Storage Service",
+      "type": "object",
+      "javaType": "org.openmetadata.catalog.type.StorageClassType",
+      "enum": [
```

Comment:
I'm not fully satisfied with defining these here.

Alternative approach:
Make storage class as a string. 

In data (json/schema/data/storageClasses/*.json) define all combinations of storage Classes

json/schema/data/storageClasses/gcs/nearline.json
```
{
  "storageServiceType": "GCS",
  "storageClassType": "Nearline"
}
```

@sureshms @harshach Any thoughts on this one?

Reply:
The storage class is per object in AWS while the storage service type is per group of buckets. I would use a copy of the AWS Java SDK enum. https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/model/StorageClass.html

Reply:
This is a great suggestion.

GCS
https://googleapis.dev/java/google-cloud-storage/latest/index.html

Azure
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AccessTier.java